addon/bootstrap.js

```diff
@@ -86,10 +89,17 @@ async function startup(addonData, reason) {
   await studyUtils.startup({reason});
 
   console.log(`info ${JSON.stringify(studyUtils.info())}`);
-  // studyUtils.endStudy("user-disable");
+  // get the homepage and run RAPPOR
   let eLTDHomepages = getHomepage();
```

Comment:
Can you move all the homepage functions and logic to a separate module and just invoke module.reportValue() from here? For example HomepageStudy.jsm?

The function could return a different success/failure value so that we could end the study with the appropriate reason here.